windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -731,19 +739,36 @@ UwbConnector::NotificationListenerStop()
 ::uwb::RegisteredCallbackToken*
 UwbConnector::RegisterDeviceEventCallbacks(std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks> callbacks)
 {
-    m_deviceEventCallbacks = callbacks;
+    {
+        std::lock_guard presenceLock{ m_callbacksPresentConditionVariableGate };

```

Comment:
Consider using `std::scoped_lock` instead of `std::lock_guard`, as `std::scoped_lock` is the newer and superior locking mechanism

Reply:
If I understand correctly then scoped_lock behaves the same as lock_guard but also has support for grabbing multiple mutexes in succession in the same line. Since we're also considering combining the two mutexes, I think I'll stick with lock_guard. (Maybe there's less overhead)